---
title: "Trumpet Report"
output: 
  html_document: 
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 1 Introduction to Trumpet

MeRIP-seq can be considered as the marriage of the methylation of DNA immunization coprecipitation(MeDIP) technique, RNA binding protein immunization coprecipitation(RIP) and RNA sequencing technique. It has the high precision to detect RNA-methylation within the scope of whole genome.

In MeRIP-seq - as in all other NGS applications - quality control is essential. However, to our knowledge, no metrics or software are used to assess the MeRIP-seq experiment data recently. So, we develop a R package that has some metrics to assess the MeRIP-seq data quality easily and visualize the assessment result. `Trumpet`, the R package for quality assessment of MeRIP-seq data, is integrated in Bioconductor. 

`Trumpet` contains basic modules to evaluate sequence quality such as, statistics reads count and read alignment summary, the reads coverage distribution. The specific modules are used to evaluate the enrichment signal strength using SES arithmetic and c-test method. The last modules are mainly used to show the similarity of IP and Input biological replicates and identify some outlier samples applying the methods of hierarchical clustering and mean-SD relationship. The quality assessment report will be shown in the form of tables and figures.

# 2 Start using Trumpet

## 2.1 Input data for quality assessment

The MeRIP-seq data for quality assessment should include IP(the methylated RNA immunoprecipitation samples) biological replicates and Input(RNA-seq data samples) control biological replicates in one or two conditions, which means that you can evaluate the IP, Input samples or IP, Input and contrast IP, contrast Input samples. These samples should be BAM format files with your alignment reads.

## 2.2 Get reads count from your samples

The following part will extract the reads alignment summary and reads count from your input samples for subsequent quality assessment. It must input smaples' paths(e.g. the path of `IP_BAM, Input_BAM`) and reference genome that can be a gene annotation gtf file or txdb file.

```{r loadLibrary,message=FALSE,warning=FALSE,echo=FALSE}
library(Trumpet)
setwd(outparam_dir)
load("parameter.Rdata")
result<-.get_readscount(IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,GENE_ANNO_GTF, GENOME,UCSC_TABLE_NAME,TXDB,sample_size)

```

# 3 Basic quality assessment module using Trumpet package

## 3.1 Read alignment summary and partial reads count statistics

In this module, we can get the reads alignment summary information and check the percentages of partial reads count of every sample in in each transcript and each bin. This metric can detect the sequencing depth and show the state of RNA transcription abundance. The result can return six tables.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE}
stats_result<-.stats_readscount(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM)
transform_table<-stats_result[[1]]
reads_summary<-stats_result[[2]]
p_trans<-stats_result[[3]]
pt_trans<-stats_result[[4]]
p_bin<-stats_result[[5]]
pt_bin<-stats_result[[6]]
kable(transform_table,caption='Input BAM information',align = 'c')
kable(reads_summary,caption='Read alignment summary',align = 'c')
kable(p_trans,caption='The percentage of IP transcript with reads count in different ranges',align = 'c')
kable(pt_trans,caption='The percentage of Input transcript with reads count in different ranges',align = 'c')
kable(p_bin,caption='The percentage of IP bin with reads count in different ranges',align='c')
kable(pt_bin,caption='The percentage of Input bin with reads count in different ranges',align='c')
```

**Note: The explanation of the column information in each table.**

**1. Table Input BAM Information**

* **file:** The path of input BAM files 

* **sample name:** The name of input BAM files added automatically

**2. Table Read Summary Information**

* **total reads#:** Total number of alignment reads in each BAM file

* **exon reads#:** The number of reads overlaped into exon region in each BAM file

* **intron reads:** The number of reads overlaped into intron region and the percentage of reads count overlaped into intron region that was calculated by $percent\_intron = \frac{intron\_reads^{\#}}{intron\_reads^{\#} + exon\_reads^{\#}}$ in each BAM file

* *$intron\_reads^{\#}$* means the number of reads overlaped into intron region and *$exon\_reads^{\#}$* means the number of reads count overlaped into exon region. 

* **5'UTR reads:** The number of reads overlaped into 5'UTR region and the percentage of reads count overlaped into 5'UTR region that was calculated by $percent\_5'UTR = \frac{5'UTR\_reads^{\#}}{5'UTR\_reads^{\#} + CDS\_reads^{\#} + 3'UTR\_reads^{\#}}$ in each BAM file.

* **CDS reads:** The number of reads overlaped into CDS region and the percentage of reads count overlaped into CDS region that was calculated by $percent\_CDS = \frac{CDS\_reads^{\#}}{5'UTR\_reads^{\#} + CDS\_reads^{\#} + 3'UTR\_reads^{\#}}$  in each BAM file.

* **3'UTR reads** : The number of reads overlaped into 3'UTR region and the percentage of reads count overlaped into 3'UTR region that was calculated by $percent\_3'UTR = \frac{3'UTR\_reads^{\#}}{5'UTR\_reads^{\#} + CDS\_reads^{\#} + 3'UTR\_reads^{\#}}$ in each BAM file.

* *$5'UTR\_reads^{\#}$* means the number of reads overlaped into 5'UTR region, *$CDS\_reads^{\#}$* means the number of reads overlaped into CDS region and *$3'UTR\_reads^{\#}$* means the number of reads overlaped into 3'UTR region.

**3. The percentage of transcripts or bins within reads count in different ranges**

* **0** : The percentage of transcripts or bins with reads count that is equal to 0 was calculated by dividing the all number of transcripts or bins of each BAM file.

* **10~100** : The percentage of transcripts or bins with reads count that is in the range between 10~100 was calculated by dividing all number of transcripts or bins of each BAM file.

## 3.2 Read coverage distribution 

This part will show the reads coverage distribution between coding exons(CDS), 5'UTR region and 3'UTR region under different reads count quantile. This module roughly reflects the preference distribution of IP samples in each gene comparing the Input samples at different transcription abundance. Such as the m6A is highly enrichment near the stop codon and 3'UTR.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=9,fig.height=12}
read_coverage<-.read_distribute(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,condition1,condition2)
```

# 4 Enrichment signal strength for quality assessment

## 4.1 Evaluate enrichment signal strength using ESES method

This module will evaluate the IP biological replicates' enrichment signal strength to identify potentially failed experiments. This function works with reads count that has been normalized in IP and Input samples. ESES method that is based on order statistics estimates the percentage of the IP samples' enrichment signal and the strength of signal by comparing the unified Input sample. It can also detect abnormal Input control samples. The assessment result will return figures and a table.

```{r  eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=10,fig.height=10}
# Graphical summary of IP enrichment and the difference between Input samples using SES method
out_SES<-.ses_evluate(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,condition1,condition2)
## Tables show enrichment signal strength
kable(out_SES)
```

**Note:The x-axis denotes the percentages of bins in the whole genome and the y-axis denotes percentages of the cumulative reads count after normalization. The black line divides the genome into two regions: the background region and the enrichment signal region. The point at which the distance of the percentage between the IP and Input is maximized shows the strength of enrichment signal in IP samples; the greater the separation between IP and Input at this point, the better the IP enrichment at appropriate sequencing depth. The parameter `Scale_factor` can stand for it.**

## 4.2 Evaluate enrichment signal strength using c-test method

C-test is used to compare two poisson means in IP and Input samples for peak calling in exomePeak R package. `ctest_evluate` can calculate the percentage of enrichment signal using c-test method at different fold-change in IP samples compared with unified Input samples and calculate the percentage of enrichment signal in unified IP samples compared with different Input samples. It will plot some figures to show the assessment result and identify problem biological replicates.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=8,fig.height=8}
# Show the figures of c-test result
out_chest<-.ctest_evluate(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,condition1,condition2)
```

# 5 Show the difference within biological replicates

## 5.1 Show the difference within biological replicates using hcluster

In order to identify the difference within IP biological replicates or the Input biological replicates, we can use the hierachical clustering method after reads count having been normalized between samples.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=7,fig.height=7}
hcluster<-.hcluster(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM)
```

## 5.2 Plot the relationship between mean and SD figures

In this work, the observed standard deviation(SD) is tightly related to the mean in each condition. The relationship should be similarity within biological replicates in one condition. Using this strategy, we can identify some abnormal samples. The value of mean and SD have been taken the logarithm.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=8,fig.height=8}
out_ms<-.ms_relation(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,condition1,condition2)
```

# 6 Compare the difference of two groups biological replicates.

This module is to identify the difference between two conditions through the mean and SD relationship curve. The IP or Input biological replicates' reads count between two condition should be normalized when using this module. Conveniently, this module can normalize the reads count between biological replicates automatically.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=7,fig.height=10}
two_ms<-.two_cond_ms(result,IP_BAM,Input_BAM,contrast_IP_BAM,contrast_Input_BAM,condition1,condition2)
```

# 7 Compare IP samples' enrichment signal to known data set.

In this part, we will compare the user's IP samples' enrichment signal to the our known experiment data set. We use the density plot to show the distribution of IP samples' signal strength that was depicted by the percentage of enrichment region, the percentage of signal reads count and scale factor.The three density distribution curve are our known experiment data. These vetical lines in each plot are user's sample(the data to be evaluated). The following plot will show the result.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=7,fig.height=14}
enrichmentdist <- .SES_enrich_DF(result, IP_BAM, Input_BAM, contrast_IP_BAM, contrast_Input_BAM)
```

**Note: In this part, we can give the reasonable range of IP samples' enrichment signal in our known experiment data set under different cell line or different condition. The percentage of enrichment region of our known data is in the range of 15% to 20%. The percentage of signal reads count of our knowing data is in the range of 90% to 95%. The Scale_factor of our known data is in the range of 0.05 to 0.15 and 0.2 to 0.25. If the user's IP samples' enrichment signal is in the range of our known experiment data set, the enrihcment degree of user's IP samples is normal.**
