---
title: "Trumpet Report"
author: "Teng Zhang"
date: "Feb 20, 2017"
output: 
  html_document: 
    toc: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# 1 Introduction to Trumpet

MeRIP-seq can be considered as the marriage of the methylation of DNA immunization coprecipitation(MeDIP) technique, RNA binding protein immunization coprecipitation(RIP) and RNA sequencing technique. It has the high precision to detect RNA-methylation within the scope of whole genome.

In MeRIP-seq - as in all other NGS applications - quality control is essential. However, to our knowledge, no metrics or software are used to assess the MeRIP-seq expriement data recently. So, we develop a R package that has some metrics to assess the MeRIP-seq data quality easily and visualize the assessment result. `Trumpet`, the R package for quality assessment of MeRIP-seq data,is integrated in Bioconductor. 

`Trumpet` contains basic modules to evaluate sequence quality such as, statistics reads count and read alignment summary, the reads coverage distribution. The specific modules are used to evaluate the enrichment signal strength using SES arithmetic and c-test method. The last modules are mainly used to show the similarity of IP and Input biological replicates and identify some outlier samples applying the methods of hierarchical cluster and mean-SD relationship. The quality assessment report will be shown in the form of tables and figures.

# 2 Getting started using Trumpet

## 2.1 Input data for quality assessment

The MeRIP-seq data for quality assessment should include IP(the methylated RNA immunoprecipitation samples) biological replicates and Input(RNA-seq data samples) control biological replicates in one or two condition, which means that you can evaluate the IP, Input samples or IP, Input and contrast IP, contrast Input samples. This samples should be BAM format  with your alignment reads. The reads count and other useful parameter have been gained from the function of **get\_readscount2** firstly. User should load the **Input_Infor.Rdata** that they need to assess.

```{r loadLibrary,message=FALSE,warning=FALSE,echo=FALSE}
library(Trumpet)
setwd(INPUT_DIR)
load("Input_Infor.Rdata")
```


# 3 Basic quality assessment module using Trumpet package

## 3.1 Read alignment summary and partial reads count statistics

In this module, we can get the read alignment summary information and check the percentages of partial reads count of every sample in in each transcript and each bin. This metric can detect the sequencing depth and show the state of RNA transcription abundance. The result can return six tables.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE}
stats_result<-stats_readscount(result)
transform_table<-stats_result[[1]]
reads_summary<-stats_result[[2]]
p_trans<-stats_result[[3]]
pt_trans<-stats_result[[4]]
p_bin<-stats_result[[5]]
pt_bin<-stats_result[[6]]
kable(transform_table,caption='Input BAM information',align = 'c')
kable(reads_summary,caption='Read summary information',align = 'c')
kable(p_trans,caption='Different readscounts percentage in IP transcript',align = 'c')
kable(pt_trans,caption='Different readscounts percentage in Input transcript',align = 'c')
kable(p_bin,caption='Different readscounts percentage in IP bin',align='c')
kable(pt_bin,caption='Different readscounts percentage in Input bin',align='c')
```

**Note: The explanation of the column information in each table.**

**1. Table Input BAM Information**

* **file** : The path of input BAM files 

* **sample_name** : The name of input BAM files to evaluate

**2. Table Read Summary Information**

* **total_reads** : Total number of alignment reads in each BAM files

* **exon_reads** : The number of reads mapping to exon region in each BAM files

* **intron_reads** : The number of reads mapping to intron region in each BAM files

* **percent_intron** : The percentage of reads count in intron was calculated by dividing the sum reads count of exon and intron.

* **UTR5_reads** :  The number of reads mapping to UTR5 region in each BAM files

* **CDS_reads** : The number of reads mapping to CDS region in each BAM files

* **UTR3_reads** : The number of reads mapping to UTR3 region in each BAM files

* **percent_UTR5** : The percentage of reads count in UTR5 was calculated by dividing the sum reads count of UTR5, CDS and UTR3.

* **percent_CDS** : The percentage of reads count in CDS was calculated by dividing the sum reads count of UTR5, CDS and UTR3.

* **percent_UTR3** : The percentage of reads count in UTR3 was calculated by dividing the sum reads count of UTR5, CDS and UTR3.

**3. Table Different Reads Count Percentage in each Transcript or bin**

* **0** : The percentage of reads count that is equal to 0 in each transcript or bin was calculated by dividing the sum reads count of each BAM file.

* **10~100** : The percentage of reads count that is in the range between 10~100 in each transcript or bin was calculated by dividing the sum reads count of each BAM file.

## 3.2 Read coverage distribution 

This part will show the reads coverage distribution of the fraction of reads mapped to coding exons, 5'UTR region, 3'UTR region in statistics. This module roughly reflects the preference distribution of IP samples in each gene comparing the Input samples at different transcription abundance. Such as the m6A is highly enrichment near the stop codon and 3'UTR.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
read_coverage<-read_distribute(result,condition1,condition2)
```

# 4 Enrichment signal strength for quality assessment

## 4.1 Evaluate enrichment signal strength using SES method

This module will evaluate the IP biological replicates enrichment signal strength to identify potentially failed experiments. This function works with reads count that has been normalized in IP and Input samples. SES method that is based on order statistics estimates the percentage of the IP data enriched signal by comparing the Input data. It can also detect abnormal Input control sample. The assessment result will return figures and tables.

```{r  eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
# Graphical summary of IP enrichment and the difference between Input samples using SES method
out_SES<-ses_evluate(result,condition1,condition2)
## Tables show enrichment signal strength
kable(out_SES[[1]])
kable(out_SES[[2]])
```

**Note: points on the x-axis correspond to percentages of the IP sample's genome, and points on the y-axis correspond to percentages of the total number of reads. The point at which the distance between the IP and Input percentages is maximized is denoted by the black line; the greater the separation between IP and Input at this point, the better the IP enrichment at appropriate sequencing depth, which the parameter `Scale_factor` can stand for**

## 4.2 Evaluate enrichment signal strength using c-test method

C-test is used to compare two poisson means in IP and Input samples for peak calling in exomePeak R package. `ctest_evluate` can calculate the percentages of enrichment signal using c-test method at different fold-change in IP samples compared with unified Input samples and calculate the percentages of enrichment signal in unified IP samples compared with different Input samples. It will plot beautiful pictures to show the assessment result and identify problem biological replicates.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
# Show the figures of c-test result
out_chest<-ctest_evluate(result,condition1,condition2)
```

# 5 Show the difference within biological replicates

## 5.1 Show the difference within one group biological replicates using hcluster

In order to identify the difference within one group of  the IP biological replicates and the Input biological replicates, we can use the hierachical cluster method after reads count having been normalized between samples.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
hcluster<-hcluster(result)
```

## 5.2 Plot the relationship between mean and SD figures

In this work, the observed SD is tightly relate to the mean in each condition. The relationship should be similarity within biological replicates in one condition. Using this strategy, we can identify some abnormal samples.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
out_ms<-ms_relation(result,condition1,condition2)
```

# 6 Compare the difference of two groups biological replicates.

This module is to identify the difference of mean and SD relationship between two conditions. The IP or Input biological replicates' reads count between two conditions should be normalized before identify this difference. Conveniently, this module can normalized the reads count between biological replicates automatically.

```{r eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,fig.width=14,fig.height=7}
two_ms<-two_cond_ms(result,condition1,condition2)
```
