---
title: "An Introduction to Trumpet Package"
author: "Teng Zhang"
date: "Feb 20, 2017"
output:
    BiocStyle::html_document:
    toc: true
---
<!--
% \VignetteEngine{knitr::rmarkdown}
% \VignetteIndexEntry{Using Trumpet}
-->

# Installation
```{r}
source("http://bioconductor.org/biocLite.R")
biocLite("Trumpet")
```

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```
```{r style,echo=FALSE,results='asis'}
BiocStyle::markdown()
```

# Introduction to Trumpet

MeRIP-seq can be considered as the marriage of the methylation of DNA immunization coprecipitation(MeDIP) technique, RNA binding protein immunization coprecipitation(RIP) and RNA sequencing technique. It has the high precision to detect RNA-methylation within the scope of whole genome.
In MeRIP-seq - as in all other NGS applications - quality control is essential. However, to our knowledge, no metrics or software are used to assess the MeRIP-seq expriement data recently. So, we develop a R package `Trumpet` that has some metrics to assess the MeRIP-seq data quality easily and visualizes the assessment result. 
The `Trumpet` package has the following metrics to evlate the MeRIP-seq data:

* Gain the statistics reads count and read alignment summary.

* Plot the the reads coverage distribution.

* Evaluate the enrichment signal strength using SES arithmetic and c-test method.

* Show the similarity of IP and Input biological replicates and identify some outlier samples applying the methods of hierarchical cluster and mean-SD relationship.

* Compare the difference between two condition (e.\,g., treated and untreated) using mean-SD relationship.

# Input data

The MeRIP-seq data for quality assessment should include IP(the methylated RNA immunoprecipitation samples) biological replicates and Input(RNA-seq data samples) control biological replicates in one or two condition, which means that you can evaluate the IP, Input samples or IP, Input and contrast IP, contrast Input samples. This samples should be bam format files with your alignment reads. After input the bam file to assessment, you can call the main function **Trumpet\_report** to generate the assessment report.

In this vignette, we will work with toy data, but we intensively recommend you assess the real MeRIP-seq data using our package that can gain the better result. If your Linux system can't generate HTML file directly, you can call the function 'get\_readscount2' to get the reads count from BAM files firstly. You can use the help commond of '?get\_readscount2' to learn how to use this function in detail. Then you can call the main function **Trumpet\_report** to generate the assessment report under Windows or Mac OS system. 

# Generate the assessment report

The main function **Trumpet\_report** in the *Trumpet* R-package is to generate the MeRIP-seq quality assessment report in HTML format. In this vignette, we will use four **control\_IP** replicates and three **control\_Input** replicates under **untreated-condition**. We will use one **treated\_IP** replicate and one **treated\_Input** as a contrast group. After input the BAM files, calling the **Trumpet\_report** function will generate the assessment report.

In order to learn about the main function **Trumpet\_report**, you can call the R commond '?Trumpet\_report'. From the help page, you can know how to use this function and every parameter that you may need in detail. Now let us firstly load the package and get the toy data (came with the package) ready and show you how to generate the MeRIP-seq quality assessment report in HTML format.

```{r loadLibrary,message=FALSE,warning=FALSE,echo=TRUE,eval=FALSE}
library(Trumpet)
f1=system.file("extdata", "IP1.bam", package="Trumpet")
f2=system.file("extdata", "IP2.bam", package="Trumpet")
f3=system.file("extdata", "IP3.bam", package="Trumpet")
f4=system.file("extdata", "IP4.bam", package="Trumpet")
IP_BAM=c(f1,f2,f3,f4)
f1=system.file("extdata", "Input1.bam", package="Trumpet")
f2=system.file("extdata", "Input2.bam", package="Trumpet")
f3=system.file("extdata", "Input3.bam", package="Trumpet")
Input_BAM=c(f1,f2,f3)
f1=system.file("extdata", "treated_IP1.bam", package="Trumpet")
contrast_IP_BAM=c(f1)
f2=system.file("extdata", "treated_Input1.bam", package="Trumpet")
contrast_Input_BAM=c(f2)
GENE_ANNO_GTF <- system.file("extdata", "hg19toy.gtf", package="Trumpet")
trumpet_report<-Trumpet_report(IP_BAM,
                               Input_BAM,
                               contrast_IP_BAM,
                               contrast_Input_BAM,
                               condition1="untreated",
                               condition2="treated",
                               GENE_ANNO_GTF=GENE_ANNO_GTF
                               )
```

If **OUTPUT\_DIR=NA** by default, the generated result of asseessment report will be saved in current path. Otherwise, you can set this parameter and the result will output the path you want.

